flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -90,6 +90,8 @@
 	/** Defines how data exchange happens - batch or pipelined */
 	private ExecutionMode executionMode = ExecutionMode.PIPELINED;
 
+	private ScheduleMode scheduleMode = ScheduleMode.EAGER;
```

Comment:
The new field may need to take part in ExecutionConfig's `equals` and `hashCode` calculation.